lib/stockboy/providers/ftp/sftp.rb

```diff
@@ -0,0 +1,66 @@
+require 'net/sftp'
+
+class SFTPAdapter
+  attr_accessor :client
+
+  def initialize(provider)
+    @provider = provider
+  end
+
+  def open
+    result = nil
+    Net::SFTP.start(@provider.host, @provider.username, password: @provider.password) do |sftp|
+      puts 'open connection'
+      @client = sftp
+      result = yield self
+      puts 'close connection'
+    end
+    result
+  end
+
+  def chdir(directory)
+    @file_dir = directory.nil? || directory.empty? ?  '/' : normalize_file_dir(directory)
+  end
+
+  def list_files
+    client.dir.entries(@file_dir).map { |e| e.name }
```

Comment:
This should be able to use the `file_name` pattern for globbing (similar to File provider) instead of listing all files and then picking one. See:

* https://github.com/avit/stockboy/blob/master/lib/stockboy/providers/file.rb#L95
* https://github.com/net-ssh/net-sftp/blob/master/lib/net/sftp/operations/dir.rb#L88